extensions/ql-vscode/src/authentication.ts

```diff
@@ -49,12 +54,12 @@ export class Credentials {
       return this.octokit;
     }
 
-    const session = await vscode.authentication.getSession(GITHUB_AUTH_PROVIDER_ID, SCOPES, { createIfNone: true });
-    this.octokit = new Octokit.Octokit({
-      auth: session.accessToken
-    });
+    await this.initializeOctokit(true);
+    // octokit shouldn't be undefined, since we've set "createIfNone: true".
+    // The following block is mainly here to prevent a compiler error.
+    if (!this.octokit) {
+      throw new Error('Failed to initialize Octokit.');
+    }
```

Comment:
Could `this.octokit` be falsy if someone clicks cancel? In this case, it's not exactly an error if octokit isn't initialized.

Assuming this is the case, I think you could have a better error message.

```suggestion
    // octokit shouldn't be undefined, since we've set "createIfNone: true".
    // The following block is mainly here to prevent a compiler error.
    if (!this.octokit) {
      throw new Error('Did not initialize Octokit.');
    }
```

Reply:
As far as I can tell, this block is basically unreachable (but I didn't know how to convince the TypeScript compiler that `this.octokit` would never actually be undefined) 🤔 

If a user clicks cancel, then the `await this.createOctokit(true)` promise is rejected and `getOctokit` errors out with `User did not consent to login`.